### PR TITLE
[metrics] channel metrics in consensus and executor

### PR DESCRIPTION
**Pull request by akichidis**
Resolves: https://github.com/MystenLabs/narwhal/issues/712

### CODE REVIEW

File: consensus/src/subscriber.rs [hunk 1]
```diff
@@ -28,11 +25,11 @@ pub struct SubscriberHandler {
     /// Receive reconfiguration update.
     rx_reconfigure: watch::Receiver<ReconfigureNotification>,
     // Channel to receive the output of consensus.
-    rx_sequence: Receiver<ConsensusOutput>,
+    rx_sequence: metered_channel::Receiver<ConsensusOutput>,
```
Comment (gdanezis) [inline consensus/src/subscriber.rs:28]: Maybe you want to make the `metered_channel` `Sender` / `Receiver` the default you use in this scope, unless you also use a lot of `mpsc`?

**Review by gdanezis — APPROVED** [review]
Awesome to be able to get metrics from these channels.

I think this is the 3rd PR that converts channels to metered channels. Is it worth putting the mpsc Send / Rec functions into the disallow list of clippy?

File: consensus/src/subscriber.rs (hunk 1 above)
Comment (akichidis) [reply consensus/src/subscriber.rs:28]: Good point, will do that!

**Comment by akichidis** [comment]
> Awesome to be able to get metrics from these channels.
> 
> I think this is the 3rd PR that converts channels to metered channels. Is it worth putting the mpsc Send / Rec functions into the disallow list of clippy?

Good idea @gdanezis . I would like to see first whether we can go as far and eliminate the use of mpsc Send/Rec in our interface with SUI (because they'll have to comply) , but we can for sure do that

File: consensus/src/metrics.rs [hunk 2]
```diff
@@ -61,3 +61,28 @@ impl Default for ConsensusMetrics {
         Self::new(default_registry())
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct ChannelMetrics {
+    /// occupancy of the channel from the `Consensus` to `SubscriberHandler`
```
Comment (huitseeker) [inline consensus/src/metrics.rs:67]: We might want to add a comment telling the reader to go look at the `tx_committed_certificates` and `tx_new_certificates` in the primary as well, as those are also relevant.

**Review by huitseeker — APPROVED** [review]
Tip top!
While I'm in favor of any measures that simplify imports, I think it's impossible to convert all channels to `metered` and that we shouldn't spend too much time on it: we use a number of ephemeral channels for cancellation, that we should not meter.

**Pull request merged**